### ISSUE TITLE
grafana-5.4.2 not working with docker?

### ISSUE THREAD

**Issue by crazy-canux** [grafana/grafana#15021]
grafana: 5.4.2
os: ubuntu16.04

t=2019-01-24T07:52:15+0000 lvl=eror msg="Server shutdown" logger=server reason="Service init failed: Migration failed err: unable to open database file"
GF_PATHS_DATA='/var/lib/grafana' is not writable.
You may have issues with file permissions, more information here: http://docs.grafana.org/installation/docker/#migration-from-a-previous-version-of-the-docker-container-to-5-1-or-later

version: '3'
services:
  grafana:
    image: harbor.sonicwall.cn:4433/captureatp-maf/grafana:5.4.2
    container_name: grafana
    hostname: grafana
    domainname: sonicwall.com
    restart: always
    networks:
      lan0:
        ipv4_address: 192.168.1.60
    volumes:
      - /opt/maf/grafana:/var/lib/grafana
      - /opt/maf/grafana/plugins:/var/lib/grafana/plugins
    environment:
      - GF_AUTH_ANONYMOUS_ENABLED=true
      - GF_USERS_DEFAULT_THEME=light
      - GF_SECURITY_ADMIN_USER=sandbox
      - GF_SECURITY_ADMIN_PASSWORD=password
    ports:
      - 3001:3000

**Comment by SamuelToh** [comment]
Suggests including the Grafana version number that you're upgrading from. 

The link in your error message could be the fix for it. Might be worth revealing the permission settings for the `/opt/maf/grafana` directory.

**Comment by crazy-canux** [comment]
It's not upgrade. Just start from 5.4.2.




    



    
    
    



    
        
            
                
                    
                        
                    
                    
                        
                            Canux CHENG
                            
                            
                                
                                    邮箱：canuxcheng@gmail.com
                                
                        
                    
                
            
        
    


Signature is customized by Netease Mail Master


On 01/24/2019 19:41, Samuel wrote: Suggests including the Grafana version number that you're upgrading from.
The link in your error message could be the fix for it. Might be worth revealing the permission settings for the /opt/maf/grafana directory.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/grafana/grafana","title":"grafana/grafana","subtitle":"GitHub repository","main_image_url":"https://github.githubassets.com/images/email/message_cards/header.png","avatar_image_url":"https://github.githubassets.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/grafana/grafana"}},"updates":{"snippets":[{"icon":"PERSON","message":"@SamuelToh in #15021: Suggests including the Grafana version number that you're upgrading from. \r\n\r\nThe link in your error message could be the fix for it. Might be worth revealing the permission settings for the `/opt/maf/grafana` directory.\r\n "}],"action":{"name":"View Issue","url":"https://github.com/grafana/grafana/issues/15021#issuecomment-457167429"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/grafana/grafana/issues/15021#issuecomment-457167429",
"url": "https://github.com/grafana/grafana/issues/15021#issuecomment-457167429",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by SamuelToh** [comment]
Maybe relax the permissions on your volume directory and try starting again. The error message is suggesting that the `Grafana` user in your container is not able to access the grafana databasefile and `var/lib/grafana` which equates `/opt/maf/grafana` on your host is not writeable

**Comment by crazy-canux** [comment]
Got the same error even delete the folder on host，




    



    
    
    



    
        
            
                
                    
                        
                    
                    
                        
                            Canux CHENG
                            
                            
                                
                                    邮箱：canuxcheng@gmail.com
                                
                        
                    
                
            
        
    


Signature is customized by Netease Mail Master


On 01/24/2019 20:02, Samuel wrote: Maybe relax the permissions on your volume directory and try starting again. The error message is suggesting that the Grafana user in your container is not able to access the grafana databasefile and var/lib/grafana which equates /opt/maf/grafana on your host is not writeable

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/grafana/grafana","title":"grafana/grafana","subtitle":"GitHub repository","main_image_url":"https://github.githubassets.com/images/email/message_cards/header.png","avatar_image_url":"https://github.githubassets.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/grafana/grafana"}},"updates":{"snippets":[{"icon":"PERSON","message":"@SamuelToh in #15021: Maybe relax the permissions on your volume directory and try starting again. The error message is suggesting that the `Grafana` user in your container is not able to access the grafana databasefile and `var/lib/grafana` which equates `/opt/maf/grafana` on your host is not writeable"}],"action":{"name":"View Issue","url":"https://github.com/grafana/grafana/issues/15021#issuecomment-457172853"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/grafana/grafana/issues/15021#issuecomment-457172853",
"url": "https://github.com/grafana/grafana/issues/15021#issuecomment-457172853",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by xlson** [comment]
@crazy-canux You are mapping in folders from your local file system into the docker container and those need to have the correct permissions. The grafana user most likely does not have write access rights to those folders and as such cant create its database files.

You can read more about how to set this up [here](http://docs.grafana.org/installation/docker/#grafana-container-using-bind-mounts).

In short, you either have to modify the permissions on the shared folders so that user with id `472` can write to them, or change what user to run the grafana process as (you can change it to be the same as who owns that folder).
